content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md

```diff
@@ -0,0 +1,413 @@
+it's a best practice to scan and alert on logs as close to real time as possible
+and to protect logs from tampering if a compromise occurs. It's important to
+reflect on the various levels of logging and identify the critical areas such as
+API endpoints.
+
+Kubernetes API audit logging can stream to a webhook and there's an example in *
+Appendix N: Webhook configuration*. Using a webhook could be a method that
+stores logs off cluster and/or centralizes all audit logs. Once logs are
+centrally managed, look to enable alerting based on critical events. Also ensure
+you understand what the baseline is for normal activities.
+
+### Alert identification
+
+While the guide stressed the importance of notifications, there is not a blanket
+event list to alert from. The alerting requirements vary based on your own
+requirements and threat model. A few examples could be:
+
+- Changes to SecurityContext of a pod
+- Updates to admission controller configs
+- Accessing certain files / URLs
+
+> TODO: Can SIG Security recommend a good sample / baseline alert events that we use in all examples moving forward? (maybe a separate issue)
+
+### Additional logging resources
+
+- [Seccomp Security Profiles and You: A Practical Guide - Duffie Cooley](https://www.youtube.com/watch?v=OPuu8wsu2Zc)
+- [TGI Kubernetes 119: Gatekeeper and OPA](https://www.youtube.com/watch?v=ZJgaGJm9NJE)
+- [Abusing The Lack of Kubernetes Auditing Policies](https://www.lacework.com/blog/hiding-in-plaintext-sight-abusing-the-lack-of-kubernetes-auditing-policies/)
+- [Enable seccomp for all workloads with a new v1.22 alpha feature](https://kubernetes.io/blog/2021/08/25/seccomp-default/)
+- [This Week in Cloud Native: Auditing / Pod Security](https://www.twitch.tv/videos/1147889860)
+
+## Upgrading and Application Security practices
+
+Kubernetes releases occur frequently and upgrade toil is a common problem for
+folks running production clusters. In addition to this, operators must
+regularly upgrade the underlying node's operating system and running
+applications. This is a best practice to ensure continued support and to reduce
+the likelihood of bugs or vulnerabilities.
+
+Kubernetes supports the 3 most recent stable releases. While each Kubernetes
+release goes through a massive gauntlet of tests before being published, some
+teams aren't comfortable running the latest stable release until some time has
+passed. No matter what version you're running ensure that patch upgrades
+happen frequently or automatically. More information can be found in
+the [version skew](https://kubernetes.io/releases/version-skew-policy/) policy
+docs.
+
+When thinking about how you'll manage node OS upgrades, consider ephemeral
+nodes. Having the ability to destroy and add nodes allows your team to respond
+quicker to node issues. In addition, having deployments that tolerate node
+instability (and a culture that encourages frequent deployments) allows for
```

Comment:
It makes sense to me as written. The polar opposite is a culture that prefers to avoid change until there is so much batched up that deploying it becomes unavoidable.